src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle',
+  color: 'red' | 'green' | 'blue',
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
 
-}
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
-export class Circle {
+  constructor(color: 'red' | 'green' | 'blue', a: number,
+    b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid sides value');
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: 'circle';
+
+  color: 'red' | 'green' | 'blue';
+
+  r: number;
+
+  constructor(color: 'red' | 'green' | 'blue', r: number) {
```

Comment:
you can access Figure types with this syntax: `Figure['typeName']` - you won't have to retype everything again